### PR TITLE
Update code samples in API reference

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: couchbase/collection.hxx [hunk 1]
```diff
@@ -348,6 +348,13 @@ public:
    * @param options the custom options
    * @return the future object with first available result, might be the active or a replica.
    *
+   * Select preferred server group in connection options:
```
Comment (DemetrisChr) [inline couchbase/collection.hxx:351]: Should this section also be added for the callback-based overload of `get_any_replica` above?

File: couchbase/collection.hxx [hunk 2]
```diff
@@ -368,6 +375,13 @@ public:
    * @note Individual errors are ignored, so you can think of this API as a best effort
    * approach which explicitly emphasises availability over consistency.
    *
+   * Select preferred server group in connection options:
+   * @snippet{trimleft} test_integration_read_replica.cxx select-preferred_server_group
+   *
+   * Restrict get_any_replica to reach nodes that belong to selected server
+   * group only:
+   * @snippet{trimleft} test_integration_read_replica.cxx get_any_replica-with-read_preference
```
Comment (DemetrisChr) [inline couchbase/collection.hxx:383]:
```suggestion
   * Restrict get_all_replicas to reach nodes that belong to selected server
   * group only:
   * @snippet{trimleft} test_integration_read_replica.cxx get_all_replicas-with-read_preference
```

File: couchbase/transactions/attempt_context.hxx [hunk 3]
```diff
@@ -50,6 +50,16 @@ public:
    * @param id The unique id of the document.
    * @return The result of the operation, which is an @ref error and a @ref
    * transaction_get_result.
+   *
+   * The example below shows how to use custom transcoder with transactions.
+   *
+   * The type should have regular transcoder to be defined. The only important
```
Comment (DemetrisChr) [inline couchbase/transactions/attempt_context.hxx:56]: > The type should have regular transcoder to be defined.

It's not very clear to me what this means. Could we phrase it differently?

Comment (avsej) [reply couchbase/transactions/attempt_context.hxx:56]: I will try to rephrase it. What I meant is that the transactions are using the same transcoders as general API, just with only exception that they should either encode types as JSON or Binary with returning corresponding flags, encoding with different set of flags is not defined.

File: couchbase/collection.hxx (hunk 1 above)
Comment (avsej) [reply couchbase/collection.hxx:351]: We do not have ready example for that case yet.

File: couchbase/transactions/attempt_context.hxx (hunk 3 above)
Comment (avsej) [reply couchbase/transactions/attempt_context.hxx]: Updated.

**Review by DemetrisChr — APPROVED** [review]

**Pull request merged**